core/cmd/shell_local.go

```diff
@@ -261,13 +261,15 @@ const ownerPermsMask = os.FileMode(0o700)
 
 // RunNode starts the Chainlink core.
 func (s *Shell) RunNode(c *cli.Context) error {
-	if err := s.runNode(c); err != nil {
+	// TODO: remove redundant drilling. BCF-2971
+	var ctx = context.Background()
```

Comment:
What does this mean?

We need to be very careful with the start-up context. It is already created later on in this method. We need connect whatever is happening here to that context tree. Can we drop this, and just pass the existing `rootCtx` to `EnsureKeys`?

Reply:
Yep, will update. thanks for the catch